benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/LocalWorker.java

```diff
@@ -250,7 +258,12 @@ private void submitProducersToExecutor(List<BenchmarkProducer> producers, KeyDis
                         final long intendedSendTime = rateLimiter.acquire();
                         uninterruptibleSleepNs(intendedSendTime);
                         final long sendTime = System.nanoTime();
-                        producer.sendAsync(Optional.ofNullable(keyDistributor.next()), payloadData).thenRun(() -> {
+                        CompletableFuture<Void> f = producer.sendAsync(Optional.ofNullable(keyDistributor.next()), payloadData);
+                        long scheduleMicros = TimeUnit.NANOSECONDS.toMicros(System.nanoTime() - sendTime);
+                        scheduleLatencyRecorder.recordValue(scheduleMicros);
+                        cumulativeScheduleLatencyRecorder.recordValue(scheduleMicros);
+                        scheduleLatencyStats.registerSuccessfulEvent(scheduleMicros, TimeUnit.MICROSECONDS);
```

Comment:
Whats the intent of doing this? AFAICT sendAsync() returns instantaneously.. seems like you want to run this logic in the beginning of sendAsync() rather than here?

.. or may be I should ask.. what is it that this metric tells us.

Reply:
it is supposed to return instantaneously but it isn't the case so I wanted to measure it. sometimes I see latency up to 20ms